### PR TITLE
imx8m name normalization

### PR DESCRIPTION
**Pull request by js731ca**
following the example of other companies, that use the chip as prefix followed by the device/board and possibly a revision; a company name is seldomly used

note that the 'remove pinmap.h' commit was done with a short script running over the files and replacing defines automatically (keeping in mind pinmap-rev-b vs -rwk) - to minimize human error

### CODE REVIEW

File: arch/arm64/boot/dts/freescale/imx8mm-crocodile-no-m4.dts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/irq.h>
-#include "crocodile-pinmap.h"
+#include "imx8mm-crocodile-pinmap.h"
```
Comment (bith3ad) [inline arch/arm64/boot/dts/freescale/imx8mm-crocodile-no-m4.dts]: Many thanks for adapting this :+1:

We should use:
$SOC-$VENODR-$PRODUCT-$REV

so:

imx8mm-leica-corocodile-revB or
imx8mm-leicageo-crocodile-revB

Also please check the compatible. This should be:

vendor,product

compatible = "leica, crocodile-revB", "leica,crocodile", "fsl,imx8mm";

or 

compatible = "leicageo,crocodile-revB", ...

File: arch/arm64/boot/dts/freescale/imx8mp-beluga.dts [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 #include <dt-bindings/phy/phy-imx8-pcie.h>
 #include "imx8mp.dtsi"
-#include "beluga-rpmsg.dtsi"
+#include "imx8mp.dtsi-beluga-rpmsg.dtsi"
```
Comment (bith3ad) [inline arch/arm64/boot/dts/freescale/imx8mp-beluga.dts:11]: This seems wrong

File: arch/arm64/boot/dts/freescale/imx8mm-crocodile-no-m4.dts [hunk 3]
```diff
@@ -111,41 +110,41 @@
 
 	pinctrl_i2c4: i2c4grp {
 		fsl,pins = <
-			CROCODILE_IMX8_I2C_4_SCL	0x400001c3
-			CROCODILE_IMX8_I2C_4_SDA	0x400001c3
+			MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL /*CROCODILE_IMX8_I2C_4_SCL*/	0x400001c3
+			MX8MM_IOMUXC_I2C4_SDA_I2C4_SDA /*CROCODILE_IMX8_I2C_4_SDA*/	0x400001c3
```
Comment (bith3ad) [inline arch/arm64/boot/dts/freescale/imx8mm-crocodile-no-m4.dts:114]: Again, many thanks for aligning it to mainline :)

One nit, the comments should be after the pad settings, so:

`MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL /*CROCODILE_IMX8_I2C_4_SCL*/	0x400001c3`

should be:

`MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL	0x400001c3   /*CROCODILE_IMX8_I2C_4_SCL*/`

This applies to all.

File: arch/arm64/boot/dts/freescale/Makefile [hunk 4]
```diff
@@ -52,6 +52,10 @@ dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls1028a-qds-9999.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8dxl-evk.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-beacon-kit.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-data-modul-edm-sbc.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mm-crocodile-no-m4.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mm-crocodile-rev-b.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mm-crocodile-rev-b-rwk.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mm-crocodile-rev-c.dtb
```
Comment (bith3ad) [inline arch/arm64/boot/dts/freescale/Makefile:58]: Where are the old dts file names?

**Review by bith3ad — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/freescale/Makefile (hunk 4 above)
Comment (js731ca) [reply arch/arm64/boot/dts/freescale/Makefile:58]: good question - was asking myself that... shouldn't they have been there already?
(only found them in the fitimage recipe...)

File: arch/arm64/boot/dts/freescale/imx8mm-crocodile-no-m4.dts (hunk 1 above)
Comment (js731ca) [reply arch/arm64/boot/dts/freescale/imx8mm-crocodile-no-m4.dts]: probably s/leicageo/hexagongeo/ instead - longish ...

**Comment by js731ca** [comment]
V2: swivel pinmap-names to the end-o-line; add 'hexagongeo-' name-prefix; refine compatible

**Comment by js731ca** [comment]
V2: swivel pinmap-names to the end-o-line; add 'hexagongeo-' name-prefix; refine compatible

**Comment by zandrey** [comment]
While strictly having no issues with this PR, I'd like that it is aligned with #122 since we need a common naming for all DTBs.

I suggest that we use `leica` instead of `hexagongeo` here.

@js731ca @Qingwu-Li Please have both PRs using the same naming convention.

**Comment by js731ca** [comment]
> I suggest that we use `leica` instead of `hexagongeo` here.

wasn't there "writing on the wall" that the license fees for "leica geosystems" won't be payed anymore in the not so distant future, and with that the name would be switched to hexagon-something?

**Comment by js731ca** [comment]
V4: 'normalized' to leica-crocodile/-beluga instead of hexagongeo

**Comment by js731ca** [comment]
@zandrey product vs project, shouldn't the devicetrees/compatibles look more like: imx8mm-hgs-ab123-rev-c

**Comment by zandrey** [comment]
> @zandrey product vs project, shouldn't the devicetrees/compatibles look more like: imx8mm-hgs-ab123-rev-c with leica->hgs and crocodile->...

Agree, let us use the naming scheme like following:
```
<soc name>-hgs-<product name>[-<revision>]
```

In addition, since you're on this part, please "normalize" also Copyright notice format and SPDX License IDs matching rules as they are referenced in [SDPX specification](https://spdx.github.io/spdx-spec/v2.3/license-matching-guidelines-and-templates/#b11-copyright-notice)

**Comment by js731ca** [comment]
ported recent PR revision as new PR into the meta-layer, since the devicetrees have been moved there recently

**Pull request closed without merging**